src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -0,0 +1,309 @@
+                 "Offset greater than file size",
+                 &response_params);
+    return;
+  }
+
+  // Load data from file as binary data
+  LOG4CXX_DEBUG(logger_, "Load binary data from file");
+  std::vector<uint8_t> bin_data;
+  if (!file_system::ReadBinaryFile(full_path, bin_data, offset_)) {
+    LOG4CXX_ERROR(logger_, "Failed to read from file: " << full_path);
+    SendResponse(false,
+                 mobile_apis::Result::GENERIC_ERROR,
+                 "Unable to read from file",
+                 &response_params);
+    return;
+  }
+
+  // Construct response message
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    response_params[strings::offset] = offset_;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+    response_params[strings::length] = length_;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::file_type)) {
+    response_params[strings::file_type] = file_type_;
+  }
+  const uint32_t crc_calculated = GetCrc32CheckSum(bin_data);
+  response_params[strings::crc32_check_sum] = crc_calculated;
+
+  LOG4CXX_DEBUG(logger_, "file_name: " << file_name_);
+  LOG4CXX_DEBUG(logger_, "file_type: " << file_type_);
+  LOG4CXX_DEBUG(logger_, "offset: " << offset_);
+  LOG4CXX_DEBUG(logger_, "length: " << length_);
+  LOG4CXX_DEBUG(logger_, "file_path: " << file_path);
+  LOG4CXX_DEBUG(logger_, "crc: " << crc_calculated);
+
+  SendResponse(true,
+               mobile_apis::Result::SUCCESS,
+               "File uploaded",
+               &response_params,
+               bin_data);
+}
+
+void GetFileRequest::on_event(const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (hmi_apis::FunctionID::BasicCommunication_GetFileFromHMI != event.id()) {
+    return;
+  }
+  const smart_objects::SmartObject& hmi_response = event.smart_object();
+  smart_objects::SmartObject response_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  std::vector<uint8_t> bin_data;
+
+  if (hmi_response[strings::msg_params].keyExists(strings::file_type)) {
+    response_params[strings::file_type] =
+        hmi_response[strings::msg_params][strings::file_type];
+  }
+
+  if (hmi_response[strings::msg_params].keyExists(strings::offset)) {
+    response_params[strings::offset] =
+        hmi_response[strings::msg_params][strings::offset];
+    offset_ = hmi_response[strings::msg_params][strings::offset].asInt();
+  }
+
+  if (hmi_response[strings::msg_params].keyExists(strings::length)) {
+    response_params[strings::length] =
+        hmi_response[strings::msg_params][strings::length];
+    length_ = hmi_response[strings::msg_params][strings::length].asInt();
+  }
+
+  if (hmi_response[strings::msg_params].keyExists(strings::file_path)) {
+    std::string full_path =
+        hmi_response[strings::msg_params][strings::file_path].asString();
+    if (!file_system::ReadBinaryFile(full_path, bin_data, offset_)) {
```

Comment:
File existence and length checks should be here as well (as they are in the `Run` function)